packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -168,9 +168,7 @@ abstract class Layer extends AbstractNode with DiagnosticableTreeMixin {
     assert(delta != 0);
     _compositionCallbackCount += delta;
     assert(_compositionCallbackCount >= 0);
-    if (parent != null) {
-      parent!._updateSubtreeCompositionObserverCount(delta);
-    }
+    parent?._updateSubtreeCompositionObserverCount(delta);
```

Comment:
This seems like a pattern we can and should lint for, maybe file a request at https://github.com/dart-lang/linter/issues/new/choose?

Reply:
You mean this exact pattern, or something like this should also trigger a lint:
```dart
if (_overlayEntry != null) { 
  _overlayEntry!.remove();
  _overlayEntry = null;
}
```

Reply:
https://github.com/dart-lang/linter/issues/4025.

Reply:
Thanks!